x509-cert/test-support/src/zlint.rs

```diff
@@ -167,17 +171,26 @@ pub fn check_certificate(pem: &[u8], ignored: &[&str]) {
         });
 
     let mut stdout = child.stdout.take().unwrap();
+
+    // read stdout in another thread
+    let output_handle = thread::spawn(move || {
+        let mut output_buf = Vec::new();
+        stdout
+            .read_to_end(&mut output_buf)
+            .expect("read zlint output");
+        output_buf
+    });
     let exit_status = child.wait().expect("get zlint status");
 
     assert!(exit_status.success(), "zlint failed");
-    let mut output_buf = Vec::new();
-    stdout
-        .read_to_end(&mut output_buf)
-        .expect("read zlint output");
 
+    let output_buf = output_handle.join().expect("stdout thread to join");
     let output: LintResult = serde_json::from_slice(&output_buf).expect("parse zlint output");
 
     assert!(output.check_lints(ignored));
+
+    std::fs::remove_file(config_path).expect("tmp file to be removed");
+    std::fs::remove_file(cert_path).expect("tmp file to be removed");
```

Comment:
That shouldn't be necessary, when `tmpdir` gets out of scope the files inside should be collected.

I guess there could have been behavior on your platform where the open FDs to config or certificate could have stop collection, but this should be solved by the scoping you added above.